modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -2334,183 +2277,111 @@ func mystopfunc() {
-        let thisSecId = thisItem[String(kSecImportItemIdentity)]  as! SecIdentity
-
-        // Get Id's Private Key, return nil id if fails
-        var thisPrivateKey : SecKey?
-        kcStatus = SecIdentityCopyPrivateKey(thisSecId, &thisPrivateKey)
-            if kcStatus != errSecSuccess {
-                print("failed to import private key from provided with error :\(kcStatus) ")
-                return nil
-            }
-
-        // Get all relevant certs [SecCertificate] from the ID's cert chain using kSecImportItemCertChain
-        let thisCertChain = thisItem[String(kSecImportItemCertChain)] as? [SecCertificate]
-
-        // Return nil Id if errors in key or cert chain at this stage
-        guard let pKey = thisPrivateKey, let pubCerts = thisCertChain else {
-            return nil
-        }
-// end::p2p-tlsid-import-from-bundled[]
-
-// tag::p2p-tlsid-store-in-keychain[]
-// STORE THE IDENTITY AND ITS CERT CHAIN IN THE KEYCHAIN
-
-        // Store Private Key in Keychain
-        let params: [String : Any] = [
-            String(kSecClass):          kSecClassKey,
-            String(kSecAttrKeyType):    kSecAttrKeyTypeRSA,
-            String(kSecAttrKeyClass):   kSecAttrKeyClassPrivate,
-            String(kSecValueRef):       pKey
-        ]
-        kcStatus = SecItemAdd(params as CFDictionary, nil)
-            if kcStatus != errSecSuccess {
-                print("Unable to store private key")
-                return nil
-            }
-       // Store all Certs for Id in Keychain:
-       var i = 0;
-       for cert in thisCertChain! {
-            let params: [String : Any] = [
-                String(kSecClass):      kSecClassCertificate,
-                String(kSecValueRef):   cert,
-                String(kSecAttrLabel):  "doco-sync-server"
-                ]
-            kcStatus = SecItemAdd(params as CFDictionary, nil)
-                if kcStatus != errSecSuccess {
-                    print("Unable to store certs")
-                    return nil
-                }
-            i=i+1
-        }
-// end::p2p-tlsid-store-in-keychain[]
-
-// tag::p2p-tlsid-return-id-from-keychain[]
-// RETURN A TLSIDENTITY FROM THE KEYCHAIN FOR USE IN CONFIGURING TLS COMMUNICATION
-do {
-    return try TLSIdentity.identity(withIdentity: thisSecId, certs: [pubCerts[0]])
-} catch {
-    print("Error while loading self signed cert : \(error)")
-    return nil
-}
-// end::p2p-tlsid-return-id-from-keychain[]
-    } // fMyGetCert
-} // cMyGetCert
-
-
-// tag::p2p-tlsid-delete-id-from-keychain[]
-
-[CBLTLSIdentity deleteIdentityWithLabel:thisSecId error: &error];
-
-// end::p2p-tlsid-delete-id-from-keychain[]
-
-
+// tag::p2p-tlsid-manage-func[]
+@interface MyGetCert1 :NSObject
 
-// end::p2p-tlsid-manage-func[]
+- (CBLTLSIdentity*) fMyGetCert;
 
+@end
 
+@implementation MyGetCert1
+
+- (CBLTLSIdentity*) fMyGetCert {
+    NSError *error = nil;
+    CBLReplicatorConfiguration *config; // FIXME: Are we sure? this is Replicator config?
```

Comment:
Looks like its P2P / URLEndpointListener sample. but the config is ReplicatorConfiguration with clientCert. Is this correct?

Reply:
It's attempting to illustrate setting the client cert authenticator - https://docs.couchbase.com/mobile/3.0.0/couchbase-lite-objc/Classes/CBLClientCertificateAuthenticator.html